server/txSender/txSender.go

```diff
@@ -88,15 +95,33 @@ func (ts *txSender) createAndSendTxs(ctx context.Context, data *sovereign.Bridge
 	txsData := ts.dataFormatter.CreateTxsData(data)
 
 	for _, txData := range txsData {
-		tx := &coreTx.FrontendTransaction{
-			Value:    "0",
-			Receiver: ts.scBridgeAddress,
-			Sender:   ts.wallet.GetBech32(),
-			GasPrice: ts.netConfigs.MinGasPrice,
-			GasLimit: 50_000_000, // todo
-			Data:     txData,
-			ChainID:  ts.netConfigs.ChainID,
-			Version:  ts.netConfigs.MinTransactionVersion,
+		var tx *coreTx.FrontendTransaction
+
+		switch {
+		case strings.HasPrefix(string(txData), registerBridgeOpsPrefix):
+			tx = &coreTx.FrontendTransaction{
+				Value:    "0",
+				Receiver: ts.scMultisigAddress,
+				Sender:   ts.wallet.GetBech32(),
+				GasPrice: ts.netConfigs.MinGasPrice,
+				GasLimit: 50_000_000, // todo
+				Data:     txData,
+				ChainID:  ts.netConfigs.ChainID,
+				Version:  ts.netConfigs.MinTransactionVersion,
+			}
+		case strings.HasPrefix(string(txData), executeBridgeOpsPrefix):
+			tx = &coreTx.FrontendTransaction{
+				Value:    "0",
+				Receiver: ts.scEsdtSafeAddress,
+				Sender:   ts.wallet.GetBech32(),
+				GasPrice: ts.netConfigs.MinGasPrice,
+				GasLimit: 50_000_000, // todo
+				Data:     txData,
+				ChainID:  ts.netConfigs.ChainID,
+				Version:  ts.netConfigs.MinTransactionVersion,
+			}
+		default:
+			log.Error("invalid tx data received", "data", string(tx.Data))
```

Comment:
string(tx.Data) is always nil if on default, needs to be string(txData)

should you just log.Err in this case? no need for any other handling? decrease the "rating" of the consensus group, because they generated something invalid.

Reply:
This is the bridge sender service, it is not part of consensus or knows about any, it cannot decrease the rating.
Theoretically, every validator should maintain his own bridge service, so as long as the validator is honest, the service will also receive correct data

Reply:
And Right, there we should have `txData`, otherwise that value is nil, good catch